jupyter_core/paths.py

```diff
@@ -9,20 +9,137 @@
 
 
 import os
+import ast
 import sys
 import stat
 import errno
+import time
+import pathlib
 import tempfile
 import warnings
+import functools
+import traceback
+import importlib
 
 from contextlib import contextmanager
 
+import entrypoints
+
+
 pjoin = os.path.join
 
 # UF_HIDDEN is a stat flag not defined in the stat module.
 # It is used by BSD to indicate hidden files.
 UF_HIDDEN = getattr(stat, 'UF_HIDDEN', 32768)
 
+# the group names for entry_points in pyproject.toml, setup.py and .cfg to
+# provide discoverable paths
+# the entrypoint target must be a list of strings to absolute paths
+JUPYTER_DATA_PATH_ENTRY_POINT = "jupyter_data_paths"
+JUPYTER_CONFIG_PATH_ENTRY_POINT = "jupyter_config_paths"
+
+# TODO: remove, once the correct strategy is decided
+JUPYTER_ENTRY_POINT_STRATEGY = os.environ.get("JUPYTER_ENTRY_POINT_STRATEGY")
+JUPYTER_ENTRY_POINT_TIMINGS = os.environ.get("JUPYTER_ENTRY_POINT_TIMINGS")
+
+
+# TODO: remove if not parsing
+# from https://github.com/pypa/setuptools/blob/23ee037d56a6d8ab957882e1a041f67924ae04da/setuptools/config.py#L19
+class StaticModule:
+    """
+    Attempt to load the module by the name
+    """
+    def __init__(self, name):
+        spec = importlib.util.find_spec(name)
+
+        with open(spec.origin) as strm:
+            src = strm.read()
+        module = ast.parse(src)
+        vars(self).update(locals())
+        del self.self
+        # add a path
+        self.__file__ = spec.origin
+
+    def __getattr__(self, attr):
+        try:
+            return next(
+                ast.literal_eval(statement.value)
+                for statement in self.module.body
```

Comment:
It seems like it should be going after the *last* literal assignment to mimic loading, where the last literal assignment wins over previous ones:

```suggestion
                for statement in reversed(self.module.body)
```

Reply:
I filed an issue in setuptools about this at https://github.com/pypa/setuptools/issues/2602

Reply:
Sure, that makes sense... though when I've abused this for `__version__` in `setup.cfg`, it's always been one of those, "this is a singleton, mess around and find out" things. 

Would want a test, but not in the "simple" examples, probably...